test/dev.ts

```diff
@@ -58,6 +59,10 @@ if (shouldStartMemoryDB) {
   await startMemoryDB()
 }
 
+// This is needed to forward the environment variables to the next process that were created after loadEnv()
+// for example process.env.MONGODB_MEMORY_SERVER_URI otherwise app.prepare() will clear them
+nextEnvImport.updateInitialEnv(process.env)
```

Comment:
Nice!! Previously I got around this by installing 2 different versions of Next.js in our monorepo. Glad you found this